dashboard/app/helpers/users_helper.rb

```diff
@@ -1,4 +1,106 @@
 module UsersHelper
+  include ApplicationHelper
+
+  # Returns the number of lines written in the current user session.
+  # return [Integer]
+  def session_lines
+    session[:lines] || 0
+  end
+
+  # Add additional lines completed in the given session
+  # @param [Integer] added_lines
+  def session_add_lines(added_lines)
+    session[:lines] = session_lines + added_lines
+  end
+
+  # Resets all user session state (level progress, lines of code, videos seen,
+  # etc.) for tests.
+  def session_reset_for_test
```

Comment:
would it make sense to either put this in the test_helper (or just in the test case class since it's only used for the test case) to separate it from the 'real' code, or, if after your next PR, you actually have to do this in the real code (eg. for the /hoc/reset or reset_session endpoints) make it work in both test and for real?